spec/abstractops.html

```diff
@@ -1442,4 +1442,190 @@ <h1>ParseTemporalYearMonthString ( _isoString_ )</h1>
         }.
     </emu-alg>
   </emu-clause>
+
+  <emu-clause id="sec-temporal-topositiveintegerorinfinity" aoid="ToPositiveIntegerOrInfinity">
+    <h1>ToPositiveIntegerOrInfinity ( _argument_ )</h1>
+    <p>
+      The abstract operation ToPositiveIntegerOrInfinity takes argument _argument_. It converts _argument_ to +&infin; or an integer _i_ for which <emu-eqn>0 &lt; i</emu-eqn>. It performs the following steps:
+    </p>
+    <emu-alg>
+      1. Let _integer_ be ? ToInteger(_argument_).
+      1. If _integer_ is *-&infin;*<sub>𝔽</sub>, then
+        1. Throw a *RangeError* exception.
+      1. If _integer_ &le; 0, then
+        1. Throw a *RangeError* exception.
+      1. Return _integer_.
+    </emu-alg>
+  </emu-clause>
+
+  <emu-clause id="sec-temporal-preparetemporalfields" aoid="PrepareTemporalFields">
+    <h1>PrepareTemporalFields ( _temporalLike_, _fieldNames_, _optionalFieldNames_ )</h1>
+    <p>
+      The abstract operation PrepareTemporalFields reads the relevant properties of an Object _temporalLike_; ensures all the required properties are present; and creates a new Object with all the properties converted to the correct type. It performs the following steps:
+    </p>
+    <emu-alg>
+      1. Assert: Type(_temporalLike_) is Object.
+      1. Let _result_ be ? OrdinaryObjectCreate(%Object.prototype%).
+      1. For each value _property_ of _fieldNames_, do
+        1. Let _value_ be ? Get(_temporalLike_, _property_).
+        1. If _value_ is *undefined*, then
+          1. If _optionalFieldNames_ does not contain _value_, then
+            1. Throw a *TypeError* exception.
+          1. Else,
+            1. Assert: the value of _property_ is in the Property column of <emu-xref href="#table-temporal-field-requirements"></emu-xref>.
+            1. Set _value_ to the corresponding Default value of the same row.
+        1. Else,
+          1. If the value of _property_ is in the Property column of <emu-xref href="#table-temporal-field-requirements"></emu-xref>, then
+            1. Let _Cast_ represent the abstract operation named by the Conversion value of the same row.
+            1. Set _value_ to ? _Cast_(_value_).
+        1. Perform ! CreateDataPropertyOrThrow(_result_, _property_, _value_).
+      1. Let _era_ be ? Get(_result_, *"era"*).
+      1. Let _eraYear_ be ? Get(_result_, *"eraYear"*).
+      1. Let _hasEra_ be the result of performing Strict Equality Comparsion _era_ === *undefined*.
+      1. Let _hasEraYear_ be the result of performing Strict Equality Comparsion _eraYear_ === *undefined*.
+      1. Let _incompleteEraSpec_ be the result of performing Strict Equality Comparsion _hasEra_ === _hasEraYear_.
```

Comment:
This seems like it could be written more intuitively, something like

```
1. If _era_ is *undefined* and _eraYear_ is not *undefined*, or if _era_ is not *undefined* and _eraYear_ is *undefined*, then
   1. Throw a *RangeError* exception.
```

Reply:
Oh, I didn't know large compound expressions were allowed for If. Somehow I got it in my head that `$metaEcmaLang` required single comparisons, will clean up!